pkg/skaffold/initializer/build/builders.go

```diff
@@ -55,11 +55,20 @@ type BuilderImagePair struct {
 	ImageName string
 }
 
+// GeneratedBuilderImagePair pairs a discovered builder with a
+// generated image name, and the path to the manifest that should be generated
+type GeneratedBuilderImagePair struct {
+	BuilderImagePair
+	ManifestPath string
+}
+
 type Initializer interface {
 	ProcessImages([]string) error
 	BuildConfig() latest.BuildConfig
 	BuilderImagePairs() []BuilderImagePair
 	PrintAnalysis(io.Writer) error
+	UnresolvedPairs() []GeneratedBuilderImagePair
```

Comment:
Can we add some doc string to the newly added public functions?

Reply:
sure